v2/internal/pkg/operator/common.go

```diff
@@ -161,8 +162,22 @@ func (o OperatorCollector) prepareD2MCopyBatch(images map[string][]v2alpha1.Rela
 			}
 
 			// add the tag for src and dest
-			switch {
 			// applies only to catalogs
+			switch {
+			// OCPBUGS-52470
+			case img.Type == v2alpha1.TypeOperatorCatalog && len(img.TargetTag) == 0 && imgSpec.IsImageByDigestOnly():
+				// find the tag for for the catalog that was mirrored via digest
+				imageIndexDir := filepath.Join(o.Opts.Global.WorkingDir, operatorCatalogsDir, imgSpec.ComponentName(), imgSpec.Digest)
+				filteredCatalogsDir := filepath.Join(imageIndexDir, operatorCatalogFilteredDir)
+				entries, err := os.ReadDir(filteredCatalogsDir)
+				if err != nil {
+					return result, fmt.Errorf("could not find directory filtered catalog image %w", err)
+				}
+				if len(entries) > 1 {
+					return result, fmt.Errorf("the directory structure for digest based catalogs is incorrect")
```

Comment:
Maybe we want to be more descriptive here for easier debugging later?
```suggestion
					return result, fmt.Errorf("digest-based catalog directory %q structure is incorrect: expected 1 entry got %d", filteredCatalogsDir, len(entries))
```